### ISSUE TITLE
As a multi-user chat room participant I want the connection to my room with nickname and password to automatically be restored when I lose and regain connection in order to avoid manually having to enter again

### ISSUE THREAD

**Issue by olabiniV2** [coyim/coyim#823]
Currently, if you lose connection and then regain it, you will get thrown out to the lobby, and you'll have to enter again. This is frustrating, but can also mean that you lose a lot of conversation if you're away from the computer for an extended amount of time.

This behavior ONLY covers rooms that are password protected. In this situation, when I regain connection, I would like Coy to automatically try to enter the room again with the same nickname and password I used before. If this fails, it's OK that I get put in the lobby to correct the problem.

From a security standpoint, the risk of keeping the room password in memory is not a deal-breaker. We do the same with other kinds of passwords, and the threat model of CoyIM is not set up to protect against attackers that can read the local process memory for Coy.

**Comment by azlancpool** [comment]
This is not working yet

**Comment by azlancpool** [comment]
Working in this issue we noticed some `GTK-Critical` warnings when switching from the `room-lobby` to the `room-main` view. After tracking those warnings we noticed that they rise when calling the `Show()` method in the `room-main` component. 
This action is necessary because the `room-main` component is not visible by default => `visible` property is `false` in the XML definition file.

By changing this property to `true` it won't be necessary to call `Show()` for these components and the switching will be handled by the `window-container` by simply setting/removing the desired component.
